extensions/mailer/deployment/src/main/java/io/quarkus/mailer/deployment/MailerProcessor.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkus.mailer.deployment;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
+import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
+import io.quarkus.mailer.impl.*;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.vertx.deployment.VertxBuildItem;
+import io.vertx.ext.mail.MailClient;
+
+public class MailerProcessor {
+
+    @BuildStep
+    AdditionalBeanBuildItem registerClients() {
+        return AdditionalBeanBuildItem.unremovableOf(MailClientProducer.class);
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem registerMailers() {
+        return AdditionalBeanBuildItem.builder()
+                .addBeanClasses(ReactiveMailerImpl.class, BlockingMailerImpl.class)
+                .build();
+    }
+
+    @BuildStep
+    void registerAuthClass(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        // We must register the auth provider used by the Vert.x mail clients
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true,
+                "io.vertx.ext.mail.impl.sasl.AuthDigestMD5",
+                "io.vertx.ext.mail.impl.sasl.AuthCramSHA256",
```

Comment:
There aren't mailers actually using these algorithms are there?  Also we should really unify this with Elytron, at some point.

Reply:
Actually there are, for example: `Cyrus`, `Dovecot`, `Oracle`, etc... just to name a few. The upstream project `vertx-mail` does not include any security related library so `Elytron` is not a dependency.

Reply:
Understood; however, we should only be using one set of SASL implementations if possible.  Is `vertx` using the standard SASL API or are these somehow natively reactive implementations?

Reply:
@pmlopes ^ can you answer?

Reply:
@cescoffier can you open an issue to deal with this? I think we need the mailer in so I will merge this PR but we need to address @dmlloyd 's comment.